nomos-da/network/core/src/protocols/replication/behaviour.rs

```diff
@@ -43,7 +43,7 @@ pub struct ReplicationBehaviour<Membership> {
     /// nomos DA subnetworks
     membership: Membership,
     /// Relation of connected peers of replication subnetworks
-    connected: HashMap<PeerId, ConnectionId>,
+    connected: HashMap<PeerId, HashSet<ConnectionId>>,
```

Comment:
uhm, so why would a peer has more than a a connection id? It should be a single connnection per behaviour right?
I understand that you change it because it wasnt working, but I would like to understand what was happening.

Reply:
Yes, you are right there should only be one connection per peer. But until we have an actual executor node, the nomos-cli is acting on behalf of a chosen node to make a dispersal.

When dispersing, the node A and nomos-cli are creating two connections to node B with the same peer id. As this is still draft, I didn't add a TODO comment, but will do that, unless you have a better idea how we could continue with happy path tests. 

If we include nomos-cli to the membership list with a distinct peer_id, then tests might sometimes fail, because sampling depends on it.

Reply:
> Yes, you are right there should only be one connection per peer. But until we have an actual executor node, the nomos-cli is acting on behalf of a chosen node to make a dispersal.
> 
> When dispersing, the node A and nomos-cli are creating two connections to node B with the same peer id. As this is still draft, I didn't add a TODO comment, but will do that, unless you have a better idea how we could continue with happy path tests.
> 
> If we include nomos-cli to the membership list with a distinct peer_id, then tests might sometimes fail, because sampling depends on it.

Ok, I understand now. Lets add a `todo` there and open an issue. We will catch up later. So for now we can go ahead and change it later. Thanks!